RI.md

```diff
@@ -0,0 +1,842 @@
+  full or partial list.
+- **pages**:
+  for Ruby or for an installed gem.
+
+Examples (output omitted):
+
+```sh
+$ ri --list # List of classes and modules.
+$ ri ruby:  # List of Ruby pages.
+```
+
+## Why `ri`?
+
+Using `ri` may have advantages over using
+the [Ruby online documentation](https://docs.ruby-lang.org/en/master):
+
+- The `ri` documentation is always available, even when you do not have internet access
+  (think: airplane mode).
+- If you are working in a terminal window, typing `ri _whatever_` (or just `ri`)
+  may be faster than navigating to a browser window and searching for documentation.
+- If you are working in an
+  [irb \(interactive Ruby\)](https://docs.ruby-lang.org/en/master/IRB.html)
+  session, you _already_ have immediate access to `ri`:
+  just type `'help'` or `'show_doc'`.
+
+## Modes
+
+There are two `ri` modes:
+
+- <i>Static mode</i>:
+  In general, `ri` responds in its static mode
+  if a _name_ is given;
+  it shows results and exits (as in the examples above).
+  See [Static Mode][1].
+- <i>Interactive mode</i>:
+  In general, `ri` enters its interactive mode
+  if no _name_ is given;
+  in interactive mode, `ri` shows results and waits for another command:
+
+    ```sh
+    $ ri
+    Enter the method name you want to look up.
+    You can use tab to autocomplete.
+    Enter a blank line to exit.
+    >>
+    ```
+
+    See [Interactive Mode][2].
+
+## Names
+
+In both modes, static and interactive,
+`ri` responds to an input _name_ that specifies what is to be displayed:
+a document, multiple documents, or other information:
+
+- Static mode (in the shell): type `'ri _name_'`;
+  examples (output omitted):
+
+    ```sh
+    $ ri File
+    $ ri IO#readlines
+    $ ri ruby:
+    ```
+
+- Interactive mode (already in `ri`): just type the _name_;
+  examples (output omitted):
+
+    ```sh
+    $ ri
+    Enter the method name you want to look up.
+    You can use tab to autocomplete.
+    Enter a blank line to exit.
+    >> File
+    >> IO#readlines
+    >> ruby:
+    ```
+
+### Names for Class and Module Documents
+
+These example `ri` commands cite names for class and module documents
+(see [details and examples][3]):
+
+| Command                      | Shows                                                      |
+|------------------------------|------------------------------------------------------------|
+| ri File                      | Document for Ruby class File.                              |
+| ri File::Stat                | Document for Ruby nested class File::Stat.                 |
+| ri Enumerable                | Document for Ruby module Enumerable.                       |
+| ri Arr                       | Document for Ruby class Array (unique initial characters). |
+| ri Nokogiri::HTML4::Document | Document for gem class Nokogiri::HTML4::Document.          |
+| ri Nokogiri                  | Document for gem module Nokogiri.                          |
+<br>
+
+If [option \\--all][4]
```

Comment:
This idiom is what I found that keeps the double-hyphen from converting to a dash.  Also, if in code block, the token would not link, which I think matters.

Reply:
I think we will see how the community reacts to this. If it's clear what the intention is, than great. Otherwise we will probably need to review the use of these.